src/sagemaker_containers/_modules.py

```diff
@@ -108,6 +109,58 @@ def install(path, capture_error=False):  # type: (str, bool) -> None
     _process.check_error(shlex.split(cmd), _errors.InstallModuleError, cwd=path, capture_error=capture_error)
 
 
+def s3_download(url, dst):  # type: (str, str) -> None
+    """Download a file from S3.
+
+    This method acts as an alias for :meth:`~sagemaker_containers.beta.framework.files.s3_download`
+    for backward-compatibility purposes.
+
+    Args:
+        url (str): the S3 URL of the file.
+        dst (str): the destination where the file will be saved.
+    """
+    _files.s3_download(url, dst)
+
+
+def download_and_install(uri, name=DEFAULT_MODULE_NAME, cache=True):
```

Comment:
```python
def download_and_install(uri, name=DEFAULT_MODULE_NAME, cache=True):
     _files.download_and_extract(name, uri, _env.code_dir)
     prepare(_env.code_dir, name)
     install(_env.code_dir)
```

Reply:
are there any potential side effects of permanently saving it in `code_dir` and using that location for caching vs saving the file in wherever it gets pip installed?

Reply:
_modules.install still installs the module using pip, the functionality was untouched https://github.com/aws/sagemaker-containers/blob/master/src/sagemaker_containers/_modules.py#L101

The file will in the pip location as well. Do you see any issues for copying the file to code dir instead of a source dir?

Reply:
I can't think of anything off the top of my head, but I also haven't looked thoroughly enough to say with certainty that changing where the file is copied to won't make a difference. Given that this PR is about preserving previous behavior, I'd rather not introduce another change if we're not sure.

Reply:
If you want to preserve behavior, although I don't see any issues in the suggested approach you can do the following as well:

```python
def download_and_install(uri, name=DEFAULT_MODULE_NAME, cache=True):
    should_use_cache = cache and exists(name)
     if not should_use_cache:
        with _files.tmpdir() as tmpdir:
        _files.download_and_extract(name, uri, tmpdir)
        prepare(tmpdir, name)
        install(tmpdir)
```

That being said, I am okay with your current implementation, besides the repeated code.

Reply:
I'm good with your second suggestion there since it preserves behavior - definitely agree about avoiding repeating code. I'll push a change with that once I get the functional tests working